bytewriter.go

```diff
@@ -0,0 +1,68 @@
+package vfsgen
+
+import (
+	"fmt"
+	"io"
+	"strings"
+)
+
+// byteWriter encodes the input as a []byte literal.
+type byteWriter struct {
+	w     io.Writer
+	ntabs int
+	err   error
+}
+
+func (r *byteWriter) Write(p []byte) (n int, err error) {
+	for len(p) > 0 {
+		s := 16
+		if s > len(p) {
+			s = len(p)
+		}
+		r.indent()
+		r.write(p[0])
+		if s > 1 {
+			for _, c := range p[1:s] {
+				r.writeAdditional(c)
+			}
+		}
+		if len(p) > 0 && s == 16 {
+			r.writeEol()
+		}
+		n += s
+		p = p[s:]
+	}
+	return n, r.err
+}
+
+func (r *byteWriter) indent() error {
+	if r.err != nil {
+		return r.err
+	}
+	_, r.err = fmt.Fprint(r.w, strings.Repeat("\t", r.ntabs))
```

Comment:
Going to my comment above - would be nice to explain why indent is needed.

Reply:
Added the comment on the struct - it outputs the byte slices adhering to formatting rules as `gofmt` would format w/o requiring `gofmt`.